clang/test/Modules/explicit-build-missing-files.cpp

```diff
@@ -50,6 +50,7 @@ int y = a2<int>;
 // CHECK: In module 'a':
 // CHECK-NEXT: a.h:1:45: error:
 
+int z = b<int>;
 // MISSING-B: could not find file '{{.*}}b.h'
 // MISSING-B-NOT: please delete the module cache
```

Comment:
This test relied on the fact that when diagnosing line 49, the `SLocEntry` for "b.h" got deserialized as a side-effect of the binary search for "a.h", thus generating the `MISSING-B` error message. Since we no longer fully deserialize the "b.h" `SLocEntry` during the binary search for "a.h", let's force its full deserialization by triggering new diagnostics pointing directly into "b.h".

Reply:
Can we use this approach to test the current change? ie. create a version of this test that doesn't deserialize b and therefore doesn't get the diagnostic? You could maybe add more files that don't get deserialized and check that none of them show up to improve the robustness of the test?

Reply:
I'm not sure. To me, this seems like an implementation detail we shouldn't test for.